### DISCUSSION TITLE
blog/https-with-selfsigned-certificates-for-your-homelab-services/

### DISCUSSION THREAD

**Discussion by giscus** [AkashRajpurohit/AkashRajpurohit#34] [Comments]
# blog/https-with-selfsigned-certificates-for-your-homelab-services/

In this article we will deep dive into understanding how we can setup HTTPS with self-signed certificates for our Homelab services.This is often required when you are running your own services and you want to access them over HTTPS.

https://akashrajpurohit.com/blog/https-with-selfsigned-certificates-for-your-homelab-services/

**Comment by Aadityajoshi151** [comment]
I was struggling with this, gave up and kept using the IP-ADDRESS:PORT format. Your guide was so simple that I was able to do it in the first try.
Thank you for this! :)

**Reply by AkashRajpurohit** [reply]
Glad you found it helpful! :)

**Comment by linux478** [comment]
extfile? What is the name of this file and where is it located?

Thank In Advance

**Reply by AkashRajpurohit** [reply]
You need to create the file, place it anywhere, does not matter.

**Comment by AirAlarm** [comment]
Thank you very much for this article. I have spend ours before I found it and it was the only one that gave the proper solution to the problem.

**Reply by AkashRajpurohit** [reply]
Glad you found it helpful :)

**Comment by paramsiddharth** [comment]
Very helpful. :) I am using this to set up SSL for my home lab. Thank you!

**Reply by AkashRajpurohit** [reply]
You're welcome 😁
